src/main/java/org/apache/sysml/api/python/SystemML.py

```diff
@@ -25,10 +25,327 @@
+    addPyDMLStmt(fnName, lhs=m, args=args, obj=obj)
+    return m
 
-class MLContext(object):
+def addPyDMLStmt(fnName, lhs=None, args=None, obj=None):
+    pydmlStmt = [getTabs()]
+    if lhs is not None:
+        pydmlStmt = pydmlStmt + [str(lhs.getID()), ' = ']
+    if obj is None:
+        pydmlStmt = pydmlStmt + [ fnName, '(']
+    else:
+        pydmlStmt = pydmlStmt + [  obj, '.', fnName, '(']
+    if args is not None:
+        pydmlStmt = pydmlStmt + args
+    MLContext.globalMLContext.pydml.append(''.join(pydmlStmt + [')\n']))
+    
+# Constructor         
+def full(init, rows=None, cols=None):
+    if rows is None or cols is None:
+        raise Exception('rows and cols are mandatory arguments')
+    scopeCheck('full')
+    return simpleAddPyDMLStmt('full', args=[str(init), ', rows=', str(rows), ', cols=', str(cols)])
+
+
+def matrix(numPyArr):
+    scopeCheck('matrix')
+    if isinstance(numPyArr, np.ndarray):
+        colNames = []
+        numCols = len(numPyArr[0,:])
+        for i in range(0, numCols):
+            colNames = colNames + [ str('C' + str(i))]
+        pdf = pd.DataFrame(numPyArr, columns=colNames)
+        df = MLContext.globalMLContext.sqlCtx.createDataFrame(pdf, colNames)
+    elif isinstance(numPyArr, pd.core.frame.DataFrame):
+        pdf = numPyArr
+        numCols = pdf.shape[1]
+        colNames = []
+        for i in range(0, numCols):
+            colNames = colNames + [ str('C' + str(i))]
+        df = MLContext.globalMLContext.sqlCtx.createDataFrame(pdf, colNames)
+    elif hasattr(numPyArr, '_jdf'):
+        df = numPyArr
+    else:
+        raise Exception('Unsupported input type')
+    MLContext.globalMLContext.df = df
+    m = Matrix(0)
+    MLContext.globalMLContext.registerInput(m.getID(), df)
+    addPyDMLStmt('load', lhs=m, args=['\" \", format=\"csv\"'])
+    return m
+
+def dot(m1, m2):
+    return m1.dot(m2) # Matrix multiplication
+    
+def transpose(m1):
+    return m1.transpose()
+    
+def mlprint(*args, **kwargs):
+    #try:
+    #    if MLContext.globalMLContext is None:
+    #        return __builtin__.print(*args, **kwargs)
+    #except AttributeError:
+    #    pass
+    if len(kwargs) > 0:
+        raise Exception('named argument not supported for mlprint inside context')
+    if len(args) > 0:
+        pydmlStmt = [getTabs(), 'print(', toString(args[0])]
+        for i in range(1, len(args)):
+            pydmlStmt = pydmlStmt + [' + ', toString(args[i])]
+        MLContext.globalMLContext.pydml.append(''.join(pydmlStmt + [')\n']))
+    else:
+            raise Exception('print cannot be empty')
+
+def toString(val):
+    if isinstance(val, Scalar):
+        return val.getID()
+    elif isinstance(val, Matrix):
+        return 'toString(' + val.getID() + ')'
+    else:
+        return str(val)
+        
+@contextmanager
+def parfor(s, start, end, check=None, par=None, mode=None, taskpartitioner=None, tasksize=None, datapartitioner=None, resultmerge=None, opt=None):
```

Comment:
What is your opinion on making this a generator? It feels more natural to write

```python
for x in parfor(s, start, end)
```

rather than a `with`?